java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSnapLinksIT.java

```diff
@@ -0,0 +1,219 @@
+
+        MantaHttpHeaders headers = new MantaHttpHeaders();
+        headers.setContentType("application/json");
+        mantaClient.put(path, testData, headers);
+
+        final String linkPath = testPathPrefix + UUID.randomUUID() + ".json";
+        skipSnapLinkTest(linkPath, path);
+        final String linkContent = mantaClient.getAsString(linkPath);
+        Assert.assertEquals(linkContent, testData);
+    }
+
+    // TEST UTILITY METHODS
+
+    private void moveDirectoryWithContents(final String source, final String destination) throws IOException {
+        mantaClient.putDirectory(source);
+
+        mantaClient.putDirectory(source + "dir1");
+        mantaClient.putDirectory(source + "dir2");
+        mantaClient.putDirectory(source + "dir3");
+
+        mantaClient.put(source + "file1.txt", TEST_DATA);
+        mantaClient.put(source + "file2.txt", TEST_DATA);
+        mantaClient.put(source + "dir1/file3.txt", TEST_DATA);
+        mantaClient.put(source + "dir1/file4.txt", TEST_DATA);
+        mantaClient.put(source + "dir3/file5.txt", TEST_DATA);
+
+        mantaClient.move(source, destination);
+        boolean newLocationExists = mantaClient.existsAndIsAccessible(destination);
+        Assert.assertTrue(newLocationExists, "Destination directory doesn't exist: "
+                + destination);
+
+        MantaObjectResponse headDestination = mantaClient.head(destination);
+
+        boolean isDirectory = headDestination.isDirectory();
+        Assert.assertTrue(isDirectory, "Destination wasn't created as a directory");
+
+        Long resultSetSize = headDestination.getHttpHeaders().getResultSetSize();
+        Assert.assertEquals(resultSetSize.longValue(), 5L,
+                "Destination directory doesn't have the same number of entries as source");
+
+        Assert.assertTrue(mantaClient.existsAndIsAccessible(destination + "file1.txt"));
+        Assert.assertTrue(mantaClient.existsAndIsAccessible(destination + "file2.txt"));
+        Assert.assertTrue(mantaClient.existsAndIsAccessible(destination + "dir1"));
+        Assert.assertTrue(mantaClient.existsAndIsAccessible(destination + "dir2"));
+        Assert.assertTrue(mantaClient.existsAndIsAccessible(destination + "dir3"));
+        Assert.assertTrue(mantaClient.existsAndIsAccessible(destination + "dir1/file3.txt"));
+        Assert.assertTrue(mantaClient.existsAndIsAccessible(destination + "dir1/file4.txt"));
+        Assert.assertTrue(mantaClient.existsAndIsAccessible(destination + "dir3/file5.txt"));
+
+        boolean sourceIsDeleted = !mantaClient.existsAndIsAccessible(source);
+        Assert.assertTrue(sourceIsDeleted, "Source directory didn't get deleted: "
+                + source);
+    }
+
+    private void skipSnapLinkTest(final String linkPath, final String objectPath) throws IOException, MantaClientHttpResponseException {
+        try {
+            mantaClient.putSnapLink( linkPath, objectPath, null);
+        } catch (MantaClientHttpResponseException e) {
```

Comment:
```
catch (MantaClientHttpResponseException e) {
          if (e.getServerCode().equals(SNAPLINKS_DISABLED_ERROR)) {
                final String message =
                        "This integration-test class can't be run since SnapLinks" +
                                "have been disabled for this account";
                LOG.warn(message);
                throw new SkipException(message, e);
            }
         throw e;
    }
```